src/tree/updater_quantile_hist.cc

```diff
@@ -160,18 +167,28 @@ void QuantileHistMaker::Builder::BuildHistogramsLossGuide(
 
 void QuantileHistMaker::Builder::AddHistRows(int *starting_index, int *sync_count) {
   builder_monitor_.Start("AddHistRows");
-
-  for (auto const& entry : nodes_for_explicit_hist_build_) {
-    int nid = entry.nid;
+  std::vector<size_t> merged_hist(nodes_for_explicit_hist_build_.size() +
+                                  nodes_for_subtraction_trick_.size());
+  for (size_t i = 0; i < nodes_for_explicit_hist_build_.size(); ++i) {
+    merged_hist[i] = nodes_for_explicit_hist_build_[i].nid;
+  }
+  for (size_t i = 0; i < nodes_for_subtraction_trick_.size(); ++i) {
+    merged_hist[nodes_for_explicit_hist_build_.size() + i] =
+    nodes_for_subtraction_trick_[i].nid;
+  }
```

Comment:
We can use ```qexpand_depth_wise_``` because it contains all nodes to build already

Reply:
No such vector for `lossguide`. But `AddHistRows` is common for both methods.